### PR TITLE
semantic search urls

### PR DESCRIPTION
**Pull request by joelhooks**
This is exploring some of the ideas around "semantic URLs" for site search that are friendly to google.

Basically taking the URL and parsing into Algolia params and then back again.

### CODE REVIEW

File: src/lib/search-url-builder.ts [hunk 1]
```diff
@@ -9,3 +10,71 @@ export const buildTitleFromUrl = (url: string) => {
...
+}
+
+// Returns a name from the category slug.
+// The "+" are replaced by spaces and other
+// characters are decoded.
+function getCategoryName(slug) {
+  return slug.split('+').map(decodeURIComponent).join(' ')
+}
+
+export const createUrl = (searchState) => {
```
Comment (joelhooks) [inline src/lib/search-url-builder.ts:25]: all of this gets a little gross, but it has tests ;)

File: src/pages/search/[[...all]].tsx [hunk 2]
```diff
@@ -35,17 +35,15 @@ export default function SearchIndex({initialSearchState, resultsState}) {
...
+      const href = searchStateToUrl(searchState)
 
       //this is all f'd up. The general idea is to build up SEO friendly URLs for search
       //where the url would be broken up like `/react/hooks/courses/by/kent+c+dodds` which is 100%
       //possible but also fairly nuanced and complex 😅
 
-      router.push(href, href, {
-        shallow: true,
-      })
+      // router.push(href, href, {
```
Comment (joelhooks) [inline src/pages/search/[[...all]].tsx:44]: killed this for testing purposes

**Comment by jaredpalmer** [comment]
My suggestion is to kill all the slashes and use a different delimiter. The goal is just and English phrase slugified

**Comment by jaredpalmer** [comment]
My suggestion is to kill all the slashes and use a different delimiter. The goal is just and English phrase slugified

**Comment by joelhooks** [comment]
@jaredpalmer I used `~~t~~` instead, where `~~` is a replacement for the `/`

Is that any better? we've got `type`, `tag`, `query`, and `instructor` currently as the facets...

**Comment by joelhooks** [comment]
actually I like `--` better A E S T H E T I C A L L Y

**Comment by jaredpalmer** [comment]
The closer to English the better. 

react-and-amplify
videos-by-dan-abramov
react-and-redux-videos-by-kent-c-dodds
react-hooks-and-redux-and-netlify

File: src/pages/s/[[...all]].tsx [hunk 3]
```diff
@@ -56,17 +54,18 @@ export default function SearchIndex({initialSearchState, resultsState}) {
     createURL,
     onSearchStateChange,
   }
+
   return (
     <div>
+      <NextSeo noindex={!isEmpty(searchState.query)} title={pageTitle} />
```
Comment (joelhooks) [inline src/pages/s/[[...all]].tsx:60]: if the user has typed in a query we don't want to index anything. We only index when they have applied certain refinements and no query.

**Pull request merged**